forge/src/executor/inspector/logs.rs

```diff
@@ -96,9 +96,9 @@ where
     ) -> (Return, Gas, Bytes) {
         if call.contract == *HARDHAT_CONSOLE_ADDRESS {
             let (status, reason) = self.hardhat_log(call.input.to_vec());
-            (status, Gas::new(0), reason)
+            (status, Gas::new(call.gas_limit), reason)
```

Comment:
@draganrakita it seems like `Gas::new` now needs to be providing the full gas limit per call, vs gas used